### ISSUE TITLE
[Feature Request] - Get request parameters when the handler callback triggered

### ISSUE THREAD

**Issue by lanma** [Yoctol/bottender#211]
I want to get some information like that req.hostname in express when the onEvent() callback triggered. Any easy way?

**Comment by chentsulin** [comment]
Hi @lanma, 
unfortunately it's not possible to access req on the context now. What is your use case here?

**Comment by lanma** [comment]
Hi @chentsulin , when I send image to user, I need to generate the image url with hostname. And my project may be hosted with ngrok. I used req.hostname to generate the image url before so I don't need to modify the host setting in my project.

**Comment by chentsulin** [comment]
I would save hostname to someplace, for example:

```js
let hostname;

// Express Middleware
server.use((req, res, next) => {
  hostname = req.hostname;
  next();
});
```

Bot should be framework-agnostic, so It's hard to access express-specific things in it.

**Comment by lanma** [comment]
@chentsulin I think that informations in http request headers are common in every chat platform / server framework. Please consider to pass these request header informations to context.

**Comment by chentsulin** [comment]
There are some more protocols or ways to host a bot, for example Slack RTM (WebSocket)、Telegram Loog Polling, ConsoleBot or Voice. Still not sure we should use http info to define a bot.

**Comment by lanma** [comment]
@chentsulin OK, maybe context.event.isHttp and context.event.httpRequestHeader. keep null if not exist.

**Comment by tw0517tw** [comment]
IMO, bot handlers should focus on incoming **events**, not **requests**, so I don't prefer put request info into handlers. Also, the only client to bot servers is the platform provider, there is not much difference between requests.

In your case, the `hostname` you need is what you have provided to platform provider, so you can just handle it on your side. If you want to get it from ngrok, you can use their API (like what [we have done](https://github.com/Yoctol/bottender/blob/master/src/cli/shared/getWebhookFromNgrok.js) in our CLI tool). After all, you still need to set different configs between development and production.

**Comment by chentsulin** [comment]
@lanma 

```js
context.requestContext
```

is now available in `v0.15.0-rc.1`.
